cpp/include/cudf/round.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column.hpp>
+
+namespace cudf {
+
+/**
+ * @addtogroup transformation_unaryops
+ * @{
+ * @file
+ * @brief Column APIs for round
+ */
+
+/**
+ * @brief Different rounding methods for `cudf::round`
+ */
+enum class rounding_method : int32_t { HALF_UP };
+
+/**
+ * @brief Rounds all the values in a column to the specified @p decimal_places
+ *
+ * `cudf::round` currently supports HALF_UP rounding for integer and floating point numbers.
+ * When @p decimal_places is positive, rounds to the corresponding number of decimal places.
+ * When @p decimal_places is negative, rounds by removing @p decimal_places digits left of the
```

Comment:
This is not correct, it doesn't remove digits. It rounds to the nearest multiple of `10^(abs(decimal_places))`.

```suggestion
 * When @p decimal_places is negative, rounds to the nearest multiple of `10^(abs(decimal_places)`.
```

This leads me to wonder why is `decimal_places` negated?  Why is negative a positive power of 10, and positive a negative power of 10? And you also wouldn't have to explain this in terms of positive and negative, you could just say "rounds to the nearest multiple of `10^decimal_places)`". After all, that's what a decimal place is!

Reply:
Fixed :)

Reply:
I don't think it is -- you only changed the comment. The decimal_places is still negated -- positive `decimal_places` indicates negative power of 10.

Reply:
Oh ... I misunderstood and thought you only wanted the comment to change. 
...
That being said I was curious to see if removing the `std::abs()` and having this in the code but it failed two of the tests cases:
```
[ RUN      ] RoundTestsFloatingPointTypes/0.SameSignificatDigits
../../../../tests/utilities/column_utilities.cu:212: Failure
Failed
first difference: lhs[0] = 988000, rhs[0] = 987999.938
```
and 
```
[ RUN      ] RoundTests.DoubleMaxTest
../../../../tests/utilities/column_utilities.cu:212: Failure
Failed
first difference: lhs[0] = 1.6999999999999999e+308, rhs[0] = 1.7000000000000001e+308

```

Reply:
You wouldn't need to remove the abs, you still need that to correctly compute `n`. You need to invert all of your if/else conditions in both the floating point and integer paths...  

HOWEVER, I just looked at your Spark example above and the `numpy.around` docs, and both treat negative values as left of the decimal point, so please forget I suggested this, we should be consistent I guess (even if it is bonkers).